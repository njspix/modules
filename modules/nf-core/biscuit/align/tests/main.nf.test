nextflow_process {

    name "Test Process BISCUIT_ALIGN"
    script "../main.nf"
    process "BISCUIT_ALIGN"

    tag "modules"
    tag "modules_nfcore"
    tag "biscuit"
    tag "biscuit/align"
    tag "biscuit/index"

    test("sarscov2 - [bam, bai] - BiscuitIndex | single end") {
        setup {
            run("BISCUIT_INDEX") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                    """
                }
            }
        }

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:true ],
                            file(params.test_data['sarscov2']['illumina']['test_methylated_1_fastq_gz'], checkIfExists: true)
                           ]
                input[1] = BISCUIT_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - [bam, bai] - BiscuitIndex | paired end") {
        setup {
            run("BISCUIT_INDEX") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                    """
                }
            }
        }

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:true ],
                            file(params.test_data['sarscov2']['illumina']['test_methylated_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_methylated_2_fastq_gz'], checkIfExists: true)
                           ]
                input[1] = BISCUIT_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
