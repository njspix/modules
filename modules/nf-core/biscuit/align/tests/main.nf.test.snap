{
    "sarscov2 - [bam, bai] - BiscuitIndex | paired end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,aa845260a9fdebcb99a1b569820d6de2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,c1ce3dc967780b0dda4ff9611ccd420c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7ed3196b956bb59d7819f39bcf4aea9d"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,c1ce3dc967780b0dda4ff9611ccd420c"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,aa845260a9fdebcb99a1b569820d6de2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7ed3196b956bb59d7819f39bcf4aea9d"
                ]
            }
        ],
        "timestamp": "2024-01-23T17:15:34.682401239"
    },
    "sarscov2 - [bam, bai] - BiscuitIndex | single end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,aa845260a9fdebcb99a1b569820d6de2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,c1ce3dc967780b0dda4ff9611ccd420c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7ed3196b956bb59d7819f39bcf4aea9d"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam.bai:md5,c1ce3dc967780b0dda4ff9611ccd420c"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,aa845260a9fdebcb99a1b569820d6de2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7ed3196b956bb59d7819f39bcf4aea9d"
                ]
            }
        ],
        "timestamp": "2024-01-23T17:15:19.166695057"
    }
}